### PR TITLE
Fireball and Iceball Sparks

### PR DESCRIPTION
**Pull request by afrey15**
## Number of Issue Fixed
- #90 #97 #120 

### Changes Proposed in the Pull Request:
- Added spark animation to fireball and iceball
- Changed several objects into particles, including `obj_mooch`, `obj_explosion`, and `obj_countdown`
- Changed countdown timer to be 2 seconds instead of 3 (still counts from 3)
- Added rectangular collision boxes that do the damage of explosions
- The collision boxes also destroy water spawners

### Notes for Reviewer
- I created a new object called `obj_particle`. I do not know whether this will work in networking. It has not been tested

### CODE REVIEW

File: objects/obj_ball/Create_0.gml [hunk 1]
```diff
@@ -7,3 +7,18 @@ Source = noone; //who created it
 arc = false; //whether or not ball will arc
 alarm[0] = 6*30 //destroy the ball after 6 seconds
 mass = 10;
+
+// Particle System
+if obj_control.animations_on = true{
```
Comment (Regulator13) [inline objects/obj_ball/Create_0.gml]:
```suggestion
if obj_control.animations_on{
```

File: objects/obj_character/Create_0.gml [hunk 2]
```diff
@@ -66,6 +66,19 @@ right_action_is_pressed = false;
 ice_is_pressed = false;
 right_action_is_pressed = false;
 
+//Particles
+if obj_control.animations_on = true{
```
Comment (Regulator13) [inline objects/obj_character/Create_0.gml]:
```suggestion
if obj_control.animations_on{
```

**Comment by Regulator13** [comment]
Um, your known issue is a deal breaker to merging this, it must be addressed before we can put this on master

**Comment by afrey15** [comment]
🤷‍♂️ worth a shot

**Comment by Regulator13** [comment]
Since explosions have an effect on the game, perhaps they are better as instances. If you're making the particle but then adding in extra code to do the effect, you may as well just have an instance

**Comment by Regulator13** [comment]
Okay I just read your issue, perhaps that is better but it'll have to be implemented before we can merge

**Comment by Regulator13** [comment]
I noticed you forgot to mention that you reduced the freeze time to 2 seconds on your changes list. An honest mistake I'm sure.

**Comment by Regulator13** [comment]
Due to the lack of backwards compatibility this pull request is being closed and a new one with the old animations still intact will be merged instead.

**Pull request closed without merging**